go/chat/unfurl/packager.go

```diff
@@ -0,0 +1,133 @@
+package unfurl
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/keybase/client/go/chat/attachments"
+	"github.com/keybase/client/go/chat/s3"
+	"github.com/keybase/client/go/chat/storage"
+
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+)
+
+type Packager struct {
+	utils.DebugLabeler
+
+	ri           func() chat1.RemoteInterface
+	store        attachments.Store
+	s3signer     s3.Signer
+	maxAssetSize int
+}
+
+func NewPackager(l logger.Logger, store attachments.Store, s3signer s3.Signer,
+	ri func() chat1.RemoteInterface) *Packager {
+	return &Packager{
+		DebugLabeler: utils.NewDebugLabeler(l, "Packager", false),
+		store:        store,
+		ri:           ri,
+		s3signer:     s3signer,
+		maxAssetSize: 2000000,
+	}
+}
+
+func (p *Packager) assetFilename(url string) string {
+	toks := strings.Split(url, "/")
+	if len(toks) > 0 {
+		return toks[len(toks)-1]
+	}
+	return "unknown.jpg"
```

Comment:
```suggestion
	return "preview.jpg"
```

Reply:
(unless this is not user facing then idc)

Reply:
No one should ever see this, its only here in case we can't MIME detect from the content of the URL.